server/battles/BattleFlowProcessor.cpp

```diff
@@ -359,36 +374,46 @@ bool BattleFlowProcessor::tryMakeAutomaticAction(const CBattleInfoCallback & bat
 			return true;
 		}
 	}
+	return false;
+}
 
+bool BattleFlowProcessor::tryActivateBerserkPenalty(const CBattleInfoCallback & battle, const CStack * next) {
 	if (next->hasBonusOfType(BonusType::ATTACKS_NEAREST_CREATURE)) //while in berserk
-	{
-		logGlobal->trace("Handle Berserk effect");
-		std::pair<const battle::Unit *, BattleHex> attackInfo = battle.getNearestStack(next);
-		if (attackInfo.first != nullptr)
-		{
-			BattleAction attack;
-			attack.actionType = EActionType::WALK_AND_ATTACK;
-			attack.side = next->unitSide();
-			attack.stackNumber = next->unitId();
-			attack.aimToHex(attackInfo.second);
-			attack.aimToUnit(attackInfo.first);
-
-			makeAutomaticAction(battle, next, attack);
-			logGlobal->trace("Attacked nearest target %s", attackInfo.first->getDescription());
-		}
-		else
 		{
-			makeStackDoNothing(battle, next);
-			logGlobal->trace("No target found");
+			logGlobal->trace("Handle Berserk effect");
+			std::pair<const battle::Unit *, BattleHex> attackInfo = battle.getNearestStack(next);
+			if (attackInfo.first != nullptr)
+				{
+					BattleAction attack;
+					attack.actionType = EActionType::WALK_AND_ATTACK;
+					attack.side = next->unitSide();
+					attack.stackNumber = next->unitId();
+					attack.aimToHex(attackInfo.second);
+					attack.aimToUnit(attackInfo.first);
+
+					makeAutomaticAction(battle, next, attack);
+					logGlobal->trace("Attacked nearest target %s", attackInfo.first->getDescription());
+				}
+			else
+				{
+					makeStackDoNothing(battle, next);
+					logGlobal->trace("No target found");
+				}
+			return true;
```

Comment:
Formatting - place bracket on next line, aligned to statement above.
```suggestion
			if (attackInfo.first != nullptr)
			{
				BattleAction attack;
				attack.actionType = EActionType::WALK_AND_ATTACK;
				attack.side = next->unitSide();
				attack.stackNumber = next->unitId();
				attack.aimToHex(attackInfo.second);
				attack.aimToUnit(attackInfo.first);

				makeAutomaticAction(battle, next, attack);
				logGlobal->trace("Attacked nearest target %s", attackInfo.first->getDescription());
			}
			else
			{
				makeStackDoNothing(battle, next);
				logGlobal->trace("No target found");
			}
			return true;
```